src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/SolutionExplorerInProcess.cs

```diff
@@ -0,0 +1,560 @@
+            {
+                return solution.GetProjectTemplate(csharpProjectTemplate, languageName);
+            }
+
+            if (string.Equals(languageName, "visualbasic", StringComparison.OrdinalIgnoreCase)
+                && (await GetVisualBasicProjectTemplatesAsync(cancellationToken)).TryGetValue(projectTemplate, out var visualBasicProjectTemplate))
+            {
+                return solution.GetProjectTemplate(visualBasicProjectTemplate, languageName);
+            }
+
+            return solution.GetProjectTemplate(projectTemplate, languageName);
+        }
+
+        private async Task<ImmutableDictionary<string, string>> GetCSharpProjectTemplatesAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var hostLocale = await GetRequiredGlobalServiceAsync<SUIHostLocale, IUIHostLocale>(cancellationToken);
+            ErrorHandler.ThrowOnFailure(hostLocale.GetUILocale(out var localeID));
+
+            var builder = ImmutableDictionary.CreateBuilder<string, string>();
+            builder[WellKnownProjectTemplates.ClassLibrary] = $@"Windows\{localeID}\ClassLibrary.zip";
+            builder[WellKnownProjectTemplates.ConsoleApplication] = "Microsoft.CSharp.ConsoleApplication";
+            builder[WellKnownProjectTemplates.Website] = "EmptyWeb.zip";
+            builder[WellKnownProjectTemplates.WinFormsApplication] = "WindowsApplication.zip";
+            builder[WellKnownProjectTemplates.WpfApplication] = "WpfApplication.zip";
+            builder[WellKnownProjectTemplates.WebApplication] = "WebApplicationProject40";
+            return builder.ToImmutable();
+        }
+
+        private async Task<ImmutableDictionary<string, string>> GetVisualBasicProjectTemplatesAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var hostLocale = await GetRequiredGlobalServiceAsync<SUIHostLocale, IUIHostLocale>(cancellationToken);
+            ErrorHandler.ThrowOnFailure(hostLocale.GetUILocale(out var localeID));
+
+            var builder = ImmutableDictionary.CreateBuilder<string, string>();
+            builder[WellKnownProjectTemplates.ClassLibrary] = $@"Windows\{localeID}\ClassLibrary.zip";
+            builder[WellKnownProjectTemplates.ConsoleApplication] = "Microsoft.VisualBasic.Windows.ConsoleApplication";
+            builder[WellKnownProjectTemplates.Website] = "EmptyWeb.zip";
+            builder[WellKnownProjectTemplates.WinFormsApplication] = "WindowsApplication.zip";
+            builder[WellKnownProjectTemplates.WpfApplication] = "WpfApplication.zip";
+            builder[WellKnownProjectTemplates.WebApplication] = "WebApplicationProject40";
+            return builder.ToImmutable();
+        }
+
+        private static string CreateTemporaryPath()
```

Comment:
could be inlined

Reply:
➡️ I'll keep it like this for now since it might be needed again.